### ISSUE TITLE
Select word is only selecting a single character

### ISSUE THREAD

**Issue by vr8hub** [Microsoft/vscode#47270]
- VSCode Version: 1.12.1
- OS Version: OSX 10.12.6

Steps to Reproduce:
1. In an editor window, position to the beginning of a word.
2. Type shift+alt+right arrow. (On the Mac that's shift+option+right). It should select the entire word, with the cursor at the end of the word.
3. On mine, it is only selecting the first character. IOW, it is moving the cursor and highlighting one character, instead of one word.
4. Subsequent shift+alt+right's select one more character at a time.

I have not changed any keyboard assignments. I searched for shift+alt+right in Keyboard Shortcuts, and got two hits: one for cursorWordSelectRight, and one for cursorColumnSelectRight(?), both with editorTextFocus. I'm guessing the cursorColumn is interfering with the cursorWord.

1. Is that correct? When two commands are assigned to the same command in the same focus, does the first one defined win?
2. Assuming this is the problem, why is VSCode shipping with conflicting keys?
3. Or should the cursorColumnSelectRight only be in effect when box-type selection is active? That is not the case with this problem. (I actually don't even know how to do that in VSCode; I haven't needed it in the time I've been playing with it.) If that's what's supposed to be happening, then this is a bug in that handling.

**Comment by roblourens** [comment]
cursorColumnSelectRight is actually shift+alt+cmd+right so there shouldn't be a conflict.

- Can you try updating to 1.22 which was just released?
- Or can you give an example of some text where you see this? It works for me.

**Comment by vr8hub** [comment]
No, there is (was) definitely a conflict. Screenshot from Keyboard Shortcuts on 1.21.1.
![screen shot 2018-04-06 at 12 09 15 am](https://user-images.githubusercontent.com/19378955/38404428-a40fa8d8-392f-11e8-8d7d-c19ac232d781.jpeg)

Updating to 1.22.1 did change the conflict.
![screen shot 2018-04-06 at 12 21 58 am](https://user-images.githubusercontent.com/19378955/38404566-92882bd4-3930-11e8-9878-033d30503f25.jpeg)
Unfortunately, it did not change the behavior. After updating and confirming the behavior did not change, I closed all files, exited the program again, then re-opened VS Code and typed cmd-N to open a new window. I typed a sentence, then tried to select a word. Behavior did not change, still only selects a character at a time.

It happens on any text. All text. Cut-and-paste this message, e.g. Put the cursor before the N in the first word (No). Type shift+option+right. It only moves one character, select/highlighting the `N`. Type it again. It moves one more character, highlighting the `o`. And so forth. Go the end of the line. Do the same thing in reverse, i.e. shift+option+left. Same behavior — one character at a time, not one word at a time. (I just did this myself, and it did exactly as described.)

This is the behavior I get on any file, any text, new or existing. Moving a word at a time (option+right/left) works correctly, it's only selecting a word at a time that doesn't work.

(Secondary question: Is there documentation somewhere on what the various "When"s mean? E.g. the difference between EditorTextFocus and textInputFocus and editorFocus and so forth? I just did a quick web search and didn't find anything, but it was a *quick* search, so I might have missed something.)

**Comment by vr8hub** [comment]
Sigh. While doing some more research I saw a couple of instances of "try it with all extensions off," which of course I should have thought of already.

I turned them all off and, of course, it worked. I then turned them on one at a time and, of course, it was the last one on the list — "Visual Studio Keymap" — that broke it. According to its GitHub (I didn't know the list was on the Contributions page until later), it has `cursorColumnSelectRight` assigned to cursorColumnSelectRight. Now, it wasn't the source of the conflict in 1.21.1, because Keyboard Shortcuts shows "Default" on both keys. But the mapping in the extension matched the conflict.

So, I know the problem, but a couple of comments/questions.
1. Is there a reason keys an extension adds aren't displayed in Keyboard Shortcuts? They really should be. We should only have to go to *one* place to see all of our keyboard shortcuts. (If that's not already a request, I'll be glad to open one.)
2. Should the extension have textInputFocus instead of editorTextFocus for that key map, like VSCode itself does? IOW, does the extension need to change the "When" to match the change in VSCode for 1.22? And given the stock VSCode has the keystroke(s), are they even needed in the extension? Shouldn't the extension only have the key mappings that VS Code doesn't have already?

I think you can close this issue. If I need to open one for either/both of the above questions, let me know. Thanks!

**Comment by roblourens** [comment]
A keybinding source is basically just telling you whether it was customized by a user or not. A keybinding from an extension does show up in the keybindings editor, and has source=Default. @sandy081 would it be possible to show in the keybindings editor which extension a keybinding came from?

**Comment by sandy081** [comment]
Yes, I think Extension can be shown.

I did not follow the complete discussion, please provide the summary of what is this issue for?

**Comment by vr8hub** [comment]
The original problem was that the "Visual Studio" keymap extension is overriding option-shift-right to be "Select column word" instead of "Select word". Because of the context it's using, that essentially caused option-shift-right to select one character at a time instead of one word at a time. (I'm typing that from memory, check the thread to make sure that's the right key.)

But I couldn't tell that it was the extension causing the problem because the keymap is shown as "Default", which looks like it's coming from VS Code itself.

**Comment by usernamehw** [comment]
TLDR; Is this a duplicate of #37034 ?
